packages/alea-frontend/pages/blog/[postId].tsx

```diff
@@ -20,20 +20,36 @@ import {
   isModerator,
 } from '@stex-react/api';
 import { MdViewer } from '@stex-react/markdown';
-import { GetServerSidePropsContext, NextPage } from 'next';
+import fs from 'fs';
+import { NextPage } from 'next';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import MainLayout from '../../layouts/MainLayout';
 
 const BlogPostPage: NextPage = ({ post }: { post: BlogPost }) => {
+  const router = useRouter();
+  const postId = router.query.postId;
   const [userInfo, setUserInfo] = useState<UserInfo>(null);
   const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
-  const router = useRouter();
+  const [blogPost, setBlogPost] = useState<BlogPost>(post);
 
   useEffect(() => {
-    getUserInfo().then(setUserInfo);
-  }, []);
+    async function fetchBlog() {
+      try {
+        const userInfoData = await getUserInfo();
```

Comment:
Move this to a separate useEffect (like it was earlier). IS there a reason why you combined the userInfo and blogPost fetch logic?